### ISSUE TITLE
[FEAT] Support class weights in the loss for image classification

### ISSUE THREAD

**Issue by liopeer** [lightly-ai/lightly-train#966]
Sub-issue of #965.

## Description

Image classification has no way to deal with imbalanced datasets. The only loss setting today is `label_smoothing`, and both losses are built without any class weighting:

- [`image_classification/train_model.py`](https://github.com/lightly-ai/lightly-train/blob/main/src/lightly_train/_task_models/image_classification/train_model.py#L146-L156)
- [`image_classification_multihead/train_model.py`](https://github.com/lightly-ai/lightly-train/blob/main/src/lightly_train/_task_models/image_classification_multihead/train_model.py#L138-L144)

Both tasks should be covered.

## Expected behavior

Add a `class_weights` argument next to the existing `label_smoothing`, following the general design in #965:

```python
class_weights: Literal["auto"] | dict[str, float] | None = None
```

Used like this:

```python
lightly_train.train_image_classification(
    out="out/my_experiment",
    model="dinov3/vitt16",
    data={...},
    model_args={"class_weights": "auto"},
    # or: model_args={"class_weights": {"cat": 1.0, "dog": 3.5}},
)
```

`"auto"` needs a different formula per classification task, because the two tasks use different losses:

- **multiclass** -> `CrossEntropyLoss(weight=...)`, using inverse class frequency scaled so the mean weight is `1.0`. This balances the classes against each other.
- **multilabel** -> `BCEWithLogitsLoss(pos_weight=...)`, using `num_images_without_class / num_images_with_class`. This is different on purpose: `pos_weight` balances positive against negative examples *within* one class, which is what multilabel training needs. #964 is a multilabel case.

Classes with zero examples must not cause a division by zero, and an unknown class name in the dictionary should raise a clear error.

Care is needed when turning the class names into a weight tensor: classes dropped with `ignore_classes` are renumbered internally, so the weights must follow that internal order.

Please add tests for both tasks and document the new argument in `docs/source/image_classification.md`.

---

If you would like to work on this, please leave a comment so we do not duplicate work.

**Comment by bferanmi806-sketch** [comment]
Hi, I’d like to work on this.

I’ve gone through #965 and the current image classification + multihead implementations. I’ll cover both multiclass and multilabel weighting, including `auto`, `ignore_classes` remapping, zero-count classes, tests, and docs.
